plugins/modules/source_control/gitlab/gitlab_project_variable.py

```diff
@@ -431,18 +431,21 @@ def main():
         supports_check_mode=True
     )
 
+    if not HAS_GITLAB_PACKAGE:
+        module.fail_json(msg=missing_required_lib("python-gitlab"), exception=GITLAB_IMP_ERR)
+
     purge = module.params['purge']
     var_list = module.params['vars']
+    state = module.params['state']
 
     if var_list:
         variables = vars_to_variables(var_list, module)
     else:
         variables = module.params['variables']
 
-    state = module.params['state']
-
-    if not HAS_GITLAB_PACKAGE:
-        module.fail_json(msg=missing_required_lib("python-gitlab"), exception=GITLAB_IMP_ERR)
+    if state == 'present':
+        if None in [x.get('value') for x in variables]:
```

Comment:
```suggestion
        if any(x['value'] is None for x in variables):
```
(`value` is always there, so no need for `get()`)